openmetadata-ui/src/main/resources/ui/src/components/DataAssets/DataAssetsHeader/DataAssetsHeader.component.tsx

```diff
@@ -196,12 +200,42 @@ export const DataAssetsHeader = ({
       });
   };
 
+  const fetchContainerParent = async (
+    parentName: string,
+    parents = [] as Container[]
+  ) => {
+    setIsBreadcrumbLoading(true);
+    try {
+      const response = await getContainerByName(parentName, 'parent');
+      const updatedParent = [response, ...parents];
+      if (response?.parent?.fullyQualifiedName) {
+        await fetchContainerParent(
+          response.parent.fullyQualifiedName,
+          updatedParent
+        );
+      } else {
+        setParentContainers(updatedParent);
+      }
+    } catch (error) {
+      showErrorToast(error as AxiosError, t('server.unexpected-response'));
+    } finally {
+      setIsBreadcrumbLoading(false);
+    }
+  };
+
   useEffect(() => {
     if (dataAsset.fullyQualifiedName && !isTourPage) {
       fetchActiveAnnouncement();
       fetchTaskCount();
     }
-  }, [dataAsset.fullyQualifiedName]);
+    const asset = dataAsset as Container;
+    if (
+      entityType === EntityType.CONTAINER &&
+      asset?.parent?.fullyQualifiedName
+    ) {
+      fetchContainerParent(asset.parent.fullyQualifiedName);
+    }
```

Comment:
```suggestion

    if (
      entityType === EntityType.CONTAINER &&
    ) {
        const asset = dataAsset as Container;
      fetchContainerParent(asset.parent.fullyQualifiedName);
    }
```
`fetchContainerParent` will handle all the cases